src/main/java/sc/fiji/bdvpg/bdv/projector/AccumulateMixedProjectorARGB.java

```diff
@@ -0,0 +1,152 @@
+{
+	public static AccumulateProjectorFactory< ARGBType > factory = new AccumulateProjectorFactory< ARGBType >()
+	{
+		@Override
+		public AccumulateMixedProjectorARGB createAccumulateProjector(
+				final ArrayList< VolatileProjector > sourceProjectors,
+				final ArrayList< Source< ? > > sources,
+				final ArrayList< ? extends RandomAccessible< ? extends ARGBType > > sourceScreenImages,
+				final RandomAccessibleInterval< ARGBType > targetScreenImage,
+				final int numThreads,
+				final ExecutorService executorService )
+		{
+			return new AccumulateMixedProjectorARGB(
+					sourceProjectors,
+					sources,
+					sourceScreenImages,
+					targetScreenImage,
+					numThreads,
+					executorService );
+		}
+	};
+
+	private final ArrayList< Source< ? > > sourceList;
+
+	public AccumulateMixedProjectorARGB(
+			final ArrayList< VolatileProjector > sourceProjectors,
+			final ArrayList< Source< ? > > sources,
+			final ArrayList< ? extends RandomAccessible< ? extends ARGBType > > sourceScreenImages,
+			final RandomAccessibleInterval< ARGBType > target,
+			final int numThreads,
+			final ExecutorService executorService )
+	{
+		super( sourceProjectors, sourceScreenImages, target, numThreads, executorService );
+		this.sourceList = sources;
+	}
+
+	@Override
+	protected void accumulate(
+			final Cursor< ? extends ARGBType >[] accesses,
+			final ARGBType target )
+	{
+		int aAvg = 0, rAvg = 0, gAvg = 0, bAvg = 0, n = 0;
+		int aAccu = 0, rAccu = 0, gAccu = 0, bAccu = 0;
+
+		int sourceIndex = 0;
+
+		final Map< SourceAndConverter, Map< String, Object > > sourceAndConverterToMetadata = BdvService.getSourceAndConverterService().getSourceAndConverterToMetadata();
+
+		for ( final Cursor< ? extends ARGBType > access : accesses )
+		{
+			final int value = access.get().get();
+			final int a = ARGBType.alpha( value );
+			final int r = ARGBType.red( value );
+			final int g = ARGBType.green( value );
+			final int b = ARGBType.blue( value );
+
+			if ( a == 0 )
+			{
+				continue;
+			}
+
+			String projectionMode = getProjectionMode( sourceIndex, sourceAndConverterToMetadata );
+
+			if ( projectionMode.equals( "Sum" ) )
+			{
+				aAccu += a;
+				rAccu += r;
+				gAccu += g;
+				bAccu += b;
+			}
+			else if ( projectionMode.equals( "Avg" ))
+			{
+				aAvg += a;
+				rAvg += r;
+				gAvg += g;
+				bAvg += b;
+				n++;
+			}
+
+			sourceIndex++;
+		}
+
+		if ( n > 0 )
+		{
+			aAvg /= n;
+			rAvg /= n;
+			gAvg /= n;
+			bAvg /= n;
+		}
+
+		aAccu += aAvg;
+		rAccu += rAvg;
+		gAccu += gAvg;
+		bAccu += bAvg;
+
+		if ( aAccu > 255 )
+			aAccu = 255;
+		if ( rAccu > 255 )
+			rAccu = 255;
+		if ( gAccu > 255 )
+			gAccu = 255;
+		if ( bAccu > 255 )
+			bAccu = 255;
+
+		target.set( ARGBType.rgba( rAccu, gAccu, bAccu, aAccu ) );
+
+	}
+
+	private String getProjectionMode( int sourceIndex, Map< SourceAndConverter, Map< String, Object > > sourceAndConverterToMetadata )
+	{
+		Source< ? > source = sourceList.get( sourceIndex );
```

Comment:
ah! Maybe we need to think in terms of metadata which is transfered and metadata which is not transfered. Like when we duplicate a SourceAndConverter, should we transfer the metadata ?

Reply:
I did not look into the duplicate method yet. Could we merge this for now and deal with this later?

Reply:
sure!